### ISSUE TITLE
Explorer: weird structure when having some existing folders and lots of non existing

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#35320]
![image](https://user-images.githubusercontent.com/900690/30973276-09e19cf0-a46d-11e7-8ac0-b855ef998171.png)

My workspace config file (the long paths actually all exist, the numbers do not):

```json
{
	"folders": [
		{
			"path": "/Users/bpasero/Desktop/end-ts"
		},
		{
			"path": "ticino-playground"
		},
		{
			"path": "/Users/bpasero/Desktop/sqlite"
		},
		{
			"path": "/Users/bpasero/Desktop/test-js"
		},
		{
			"path": "/Users/bpasero/Desktop/notifications"
		},
		{
			"path": "1"
		},
		{
			"path": "2"
		},
		{
			"path": "3"
		},
		{
			"path": "4"
		},
		{
			"path": "5"
		},
		{
			"path": "6"
		},
		{
			"path": "7"
		},
		{
			"path": "8"
		},
		{
			"path": "9"
		},
		{
			"path": "10"
		},
		{
			"path": "11"
		}
	],
	"settings": {
	}
}
```

**Comment by isidorn** [comment]
This works fine for me with the latest vscode insiders (note that I have changed that we now seperatly resolve root folders).
Note that I already have some issues for not properly displaying non-existing folders which are captured in these issues https://github.com/Microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+assignee%3Aisidorn+milestone%3A%22October+2017%22

**Comment by roblourens** [comment]
I can still make weird things happen with a similar config. Here, first one of the folder's contents were visible even though the folder arrow was collapsed. Then I scrolled and it scrolls through the same contents in a loop. 

![nov-01-2017 17-25-40](https://user-images.githubusercontent.com/323878/32304064-bb5c8018-bf29-11e7-8f83-f19a2487f223.gif)

**Comment by isidorn** [comment]
@roblourens I can not reproduce this. Can you please share a folder and the .code-workspace file for which you can always reproduce this?
Do you see something in the developer console?

**Comment by bpasero** [comment]
Maybe https://github.com/Microsoft/vscode/issues/37237 is related where I can easily reproduce issues with the tree?

**Comment by isidorn** [comment]
Should be fixed with fix for #37237. @roblourens please try to repro again, thank you

**Comment by roblourens** [comment]
I can't repro it after that fix.
